chaynHQ/bloom-frontend
# Live courses support

Add functionality and extra blocks to support a course being “live”.

We chose a simple method (see [[proposal](https://www.notion.so/Live-courses-tech-proposal-f1f0ba306d8b48d4abe2074ebb7b59b9)](https://www.notion.so/Live-courses-tech-proposal-f1f0ba306d8b48d4abe2074ebb7b59b9)) for implementing this, keeping the backend unaware of live courses, but using storyblok fields to show extra dynamic content if a course is “live”. Storyblok `live_start_date` and `live_end_date` fields drive the logic of when to show live or coming soon content.

What a “live” course means

- 1-1 chat will be available on session pages whilst a course is live
- users can join a whatsapp/telegram list for regular messages
- the frontend shows extra tags/blocks to display “live” and “coming soon” content, which provide a place for the instructions on joining a whatsapp/telegram group. you could also include the course intro video/a preview, as part of the dynamic storyblok content.


Notes
- live courses are only available to public users, not those with partner access(es)
- see storyblok Course component for the fields driving the live functionality on the frontend

To test
- open storyblok and go to a course overview page
- test using the new `live_start/end_date` `live_soon_content` `live_now_content` to see the changes being displayed. you can save the temporary changes in storyblok, but dont publish, to see the changes.
- dont forget to undo the changes you made in storyblok, emptying the live fields, before saving and republishing

<img width="1792" alt="Screenshot 2022-03-20 at 14 57 42" src="https://user-images.githubusercontent.com/11525717/159170672-d759218a-96ba-46d1-ae84-778e9422eab3.png">
<img width="1792" alt="Screenshot 2022-03-20 at 14 58 08" src="https://user-images.githubusercontent.com/11525717/159170682-95de64f4-aa50-4d82-8542-a7407737daa4.png">
